common/src/main/java/com/wynntils/features/chat/ChatCoordinatesFeature.java

```diff
@@ -68,8 +68,12 @@ private static StyledText getStyledTextWithCoordinatesInserted(StyledText styled
 
                 String match = partToReplace.getString(null, PartStyle.StyleType.NONE);
 
-                String firstPart = match.substring(0, matcher.start());
-                String lastPart = match.substring(matcher.end());
+                int matchStart = matcher.group().charAt(0) == ' ' ? matcher.start() + 1 : matcher.start();
+                int matchEnd =
+                        matcher.group().charAt(matcher.group().length() - 1) == ' ' ? matcher.end() - 1 : matcher.end();
+
+                String firstPart = match.substring(0, matchStart);
+                String lastPart = match.substring(matchEnd);
```

Comment:
I am not sure I understand what this does with relation to the regex change. Can you explain?

Reply:
In the "My location is at x, y, z" for example, the matcher matches " x, y, z " so if you just use matcher.start() it'll appear as "My location is at[x, y, z] without a space between the at and [

Reply:
Hm. It's just that I really don't like using string manipulations, it is very easy to get them to work, but it is also very easy to not cover an edge case, and cause crashes/other issues. Do you think we could fix this in the regex itself?

Reply:
Possibly, I did start with trying to fix the issue there but didn't have any luck though I'm by no means an expert in regex

Reply:
Yeah, I see. I am not sure how I can help, but I would rather get the regex straight.

Reply:
I'll give it another try, don't want to pull you away from the 1.20 port

Reply:
Think I've got it working the same just with the regex and using group 1 instead. Wouldn't mind someone else giving it a test to make sure it's not parsing unexpected things but no rush on that, it's not a major issue